rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -183,26 +187,39 @@ def _decode_curve_remove_events(
                 (
                     user_or_contract_address == event.location_label or
                     user_or_contract_address in CURVE_DEPOSIT_CONTRACTS or
-                    tx_log.topics[0] == REMOVE_LIQUIDITY_IMBALANCE
+                    tx_log.topics[0] in REMOVE_LIQUIDITY_IMBALANCE
                 )
             ):
                 event.event_type = HistoryEventType.SPEND
                 event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
                 event.counterparty = CPT_CURVE
                 event.notes = f'Return {event.balance.amount} {crypto_asset.symbol}'
                 return_event = event
-            elif (  # Withdraw receive asset
-                event.event_type == HistoryEventType.RECEIVE and
-                event.event_subtype == HistoryEventSubType.NONE and
-                event.location_label == transaction.from_address and
-                user_or_contract_address == event.location_label and
-                tx_log.address in self.curve_pools
+            if (  # Withdraw receive asset
+                tx_log.address in self.curve_pools and
+                transaction.from_address == user_or_contract_address
             ):
-                event.event_type = HistoryEventType.WITHDRAWAL
-                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
-                event.counterparty = CPT_CURVE
-                event.notes = f'Remove {event.balance.amount} {crypto_asset.symbol} from {tx_log.address} curve pool'  # noqa: E501
-                withdrawal_events.append(event)
+                notes = f'Remove {event.balance.amount} {crypto_asset.symbol} from {tx_log.address} curve pool'  # noqa: E501
+                if (  # Raw event
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.location_label == user_or_contract_address
+                ):
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.counterparty = CPT_CURVE
+                    event.notes = notes
+                    withdrawal_events.append(event)
+                elif (  # Is already decoded by aave
```

Comment:
Shouldn't this check only be done if the pool is one of the aave pools in curve?